lib/src/invite/invite_bloc.dart

```diff
@@ -0,0 +1,201 @@
+  InviteService inviteService = InviteService();
+
+  @override
+  InviteState get initialState => InviteStateInitial();
+
+  @override
+  Stream<InviteState> mapEventToState(InviteEvent event) async* {
+    if (event is CreateInviteUrl) {
+      try {
+        yield* createInviteUrl(event.message);
+      } catch (error) {
+        yield InviteStateFailure();
+      }
+    } else if (event is HandleSharedInviteLink) {
+      yield* handleSharedInviteLink();
+    } else if (event is CreateChatWithInvite) {
+      yield* createChatWithInvite(event.inviteServiceResponse);
+    }
+  }
+
+  Stream<InviteState> createInviteUrl(String message) async* {
+    InviteServiceRequest requestInviteService = await _createInviteServiceRequest(message ?? "");
+    var response = await inviteService.createInviteUrl(requestInviteService);
+    bool valid = validateHttpResponse(response);
+    if (valid) {
+      InviteServiceResponse responseInviteService = _getInviteResponse(response);
+      Map argsMap = <String, String>{
+        'title': '',
+        'path': '',
+        'mimeType': 'text/*',
+        'text': '${responseInviteService.endpoint} \n ${L10n.get(L.inviteShareText)}'
+      };
+      await platform.invokeMethod('sendSharedData', argsMap);
+      yield InviteStateSuccess();
+    } else {
+      yield InviteStateFailure(errorMessage: response.reasonPhrase);
+    }
+  }
+
+  Stream<InviteState> handleSharedInviteLink() async* {
+    String sharedLink = await _getInitialLink();
+    int startIndex = sharedLink.lastIndexOf('/') + 1;
+    String id = sharedLink.substring(startIndex);
+    if (id.isNotEmpty) {
+      Response response = await inviteService.getInvite(id);
+      bool valid = validateHttpResponse(response);
+      if (valid) {
+        InviteServiceResponse responseInviteService = _getInviteResponse(response);
+        yield InviteStateSuccess(inviteServiceResponse: responseInviteService);
+      } else {
+        String errorMessage;
+        if (response.statusCode == 404) {
+          errorMessage = L10n.get(L.inviteGetText404Error);
+        } else {
+          errorMessage = L10n.getFormatted(L.inviteGetTextGeneralErrorX, [response.reasonPhrase]);
+        }
+        yield InviteStateFailure(errorMessage: errorMessage);
+      }
+    }
+  }
+
+  Stream<InviteState> createChatWithInvite(InviteServiceResponse inviteServiceResponse) async* {
+    Context context = Context();
+    String email = inviteServiceResponse.sender.email;
+    int contactId = await context.createContact(inviteServiceResponse.sender.name, email);
+    String imageString = inviteServiceResponse.sender.image;
+    int chatId = await context.createChatByContactId(contactId);
+    _chatRepository.putIfAbsent(id: chatId);
+    _contactRepository.putIfAbsent(id: contactId);
+    if (imageString != null) {
+      var directory = await getApplicationDocumentsDirectory();
+      File file = File("${directory.path}/${email}_avatar.png");
```

Comment:
Could it be helpful to extract this "image reload part" in a helper, so we can reuse it at other places or do you think this is specific to this area?

Reply:
@Boehrsi Sorry, I'm not in that task, but... Is this hardcoded file extension `.png` expected?

Reply:
Was moved to an image utils class. Now it can be used at different parts.

Reply:
We should ask the services guys in regards to the image file type, but I think it will be any given image type (user defined). So hard coding PNG isn't a good idea here. I'm currently unable to perform one full test cycle. If I'm able I will also try the image part.